benchmarks/segmentation_layers.py

```diff
@@ -1,175 +1,329 @@
+
+        return _max_pool2d_nt
+
+    #
+    # cross_entropy
+    #
+    def cross_entropy_tensor_iter(self):
+        inputs, targets = self.get_input(return_targets=True)
+
+        def _cross_entropy_tensor_iter():
+            for a, b in zip(inputs, targets):
+                torch.nn.functional.cross_entropy(a.unsqueeze(0), b.unsqueeze(0)).squeeze(0)
+        return _cross_entropy_tensor_iter
+
+    def cross_entropy_tensor_pad(self):
+        inputs, targets = self.get_input(return_targets=True, pad=True)
+        i_stack = torch.stack(inputs)
+        t_stack = torch.stack(targets)
+
+        def _cross_entropy_tensor_pad():
+            torch.nn.functional.cross_entropy(i_stack, t_stack)
+
+        return _cross_entropy_tensor_pad
+
+    def cross_entropy_nt(self):
+        inputs, targets = self.get_input(return_targets=True, pad=True)
+        i_nt = nestedtensor.nested_tensor(inputs)
+        t_nt = nestedtensor.nested_tensor(targets)
+        def _cross_entropy_nt():
+            torch.nn.functional.cross_entropy(i_nt, t_nt)
+
+        return _cross_entropy_nt
+
+
+    #
+    # dropout
+    #
+    def dropout_tensor_iter(self):
+        inputs = self.get_input()
+
+        def _dropout_tensor_iter():
+            for t in inputs:
+                torch.nn.functional.dropout(t.unsqueeze(0)).squeeze(0)
+        return _dropout_tensor_iter
+
+    def dropout_tensor_pad(self):
+        inputs = self.get_input(pad=True)
+        stack = torch.stack(inputs)
+
+        def _dropout_tensor_pad():
+            torch.nn.functional.dropout(stack)
+
+        return _dropout_tensor_pad
+
+    def dropout_nt(self):
+        inputs = self.get_input()
+        nt = nestedtensor.nested_tensor(inputs)
+        def _dropout_nt():
+            torch.nn.functional.dropout(nt)
+
+        return _dropout_nt
+        inputs = get_input_big_diff()
+        nt = nestedtensor.nested_tensor(inputs)
+
+        def _dropout_nt_big_diff():
+            dropout(nt)
+
+        return _dropout_nt_big_diff
+
+    #
+    # interpolate
+    #
+    def interpolate_tensor_iter(self):
+        inputs = self.get_input()
+
+        def _interpolate_tensor_iter():
+            for t in inputs:
+                torch.nn.functional.interpolate(t,  t.unsqueeze(0).shape[-2])
+        return _interpolate_tensor_iter
+
+    def interpolate_tensor_pad(self):
+        inputs = self.get_input(pad=True)
+        stack = torch.stack(inputs)
+
+        def _interpolate_tensor_pad():
+            torch.nn.functional.interpolate(stack, inputs[0].unsqueeze(0).shape[-2])
+
+        return _interpolate_tensor_pad
+
+    def interpolate_nt(self):
+        inputs = self.get_input()
+        nt = nestedtensor.nested_tensor(inputs)
+        def _interpolate_nt():
+            torch.nn.functional.interpolate(nt)
+
+        return _interpolate_nt
+
+
+
+def main(args):
+    b = SegLayersBenchMark(args)
+
+    print(utils.benchmark_fn(b.relu_tensor_iter(), warmup=b.warmup))
```

Comment:
I'd set this up to only run one benchmark per command invocation. You can then specify (and list) the name of a benchmark via an argument as well.

Then you can do (for example):
```python benchmarks/segmentation_layers.py relu_tensor_pad 6 30 100 100 0.5 0.0```

Currently then you might get a result such as

```
N =  6
C =  30
H =  100
W =  100
delta =  100
warmup =  100
fn _relu_tensor_pad avg(us):   684.7129 std(us):   124.8187 num_runs: 731
```

You could print tuples (or json etc.) that also repeats the parameters (N, C, H, etc.).

Then you can turn a list of invocations such as

```
python benchmarks/segmentation_layers.py relu_tensor_pad 6 30 100 100 0.5 0.0
python benchmarks/segmentation_layers.py relu_nt 6 30 100 100 0.5 0.0
python benchmarks/segmentation_layers.py conv2d_tensor_iter 6 30 100 100 0.5 0.0
...
```

into a table that you can use for further processing.